src/test/java/edu/wisc/my/messages/data/MessagesDataITTest.java

```diff
@@ -0,0 +1,27 @@
+package edu.wisc.my.messages.data;
+
+import org.junit.Test;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.mock.env.MockEnvironment;
+
+public class MessagesDataITTest {
+
+    /**
+    * Test that the data in the incoming text file is actually valid.
+    */
+    @Test
+    public void dataIsValid() throws Exception {
```

Comment:
This is equivalent to the

```
@Test
public void dataIsValid() {
  this.messageReader.allMessages();
}
```

that this changeset would remove from `MessagesControllerTest`. The short version fulfills the "It would be very cool" TODO mentioned inline here.